src/pages/media/AttachmentModalScreen/index.tsx

```diff
@@ -6,16 +6,19 @@ import ReportAttachmentModalContent from './routes/ReportAttachmentModalContent'
 import ReportAvatarModalContent from './routes/ReportAvatarModalContent';
 import TransactionReceiptModalContent from './routes/TransactionReceiptModalContent';
 import WorkspaceAvatarModalContent from './routes/WorkspaceAvatarModalContent';
-import type {AttachmentModalScreenParams, AttachmentModalScreenProps} from './types';
+import type {AttachmentModalScreenProps, AttachmentModalScreenType} from './types';
+
+type RouteType<Screen extends AttachmentModalScreenType> = AttachmentModalScreenProps<Screen>['route'];
+type NavigationType<Screen extends AttachmentModalScreenType> = AttachmentModalScreenProps<Screen>['navigation'];
 
 /**
  * The attachment modal screen can take various different shapes. This is the main screen component that receives the route and
  * navigation props from the parent screen and renders the correct modal content based on the route.
  */
-function AttachmentModalScreen({route, navigation}: AttachmentModalScreenProps) {
+function AttachmentModalScreen<Screen extends AttachmentModalScreenType>({route, navigation}: AttachmentModalScreenProps<Screen>) {
     const attachmentsContext = useContext(AttachmentModalContext);
-    const paramsWithContext: AttachmentModalScreenParams = useMemo(() => {
-        const currentAttachment = attachmentsContext.getCurrentAttachment();
+    const paramsWithContext = useMemo(() => {
```

Comment:
You could memo the whole route here instead of descructuring it later `{...route, params: paramsWithContext}`

Reply:
Good point. Thanks for the catch!